### PR TITLE
RFC: Custom eslint plugin

### PR DESCRIPTION
**Pull request by zomars**
## What does this PR do?

Since we're expecting many code changes and we may want to start enforcing best practices and avoiding known pitfalls. It occurred to me that a good way to achieve this is by start creating some custom eslint rules specific to our code base. This PR add a very basic one that prevent importing `dayjs` from other place that `@calcom/dayjs` to prevent a common pitfall of plugin conflicts.

Some other ideas I have for custom rules are:

- Using `defaultHandler` `defaultResponder` for API endpoints.
- Prevent importing server code on client code.
- Start displaying warnings for new bare Next.js API endpoints (and nudging into creating a tRPC router instead)
- Prefer early returns
- Prefer type inference
- etc, etc.

Please test this out and reply if this something we want and also more ideas for custom rules are welcome

https://www.loom.com/share/5902789bcb5d433b95fe72047828d7af

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How should this be tested?

- [ ] Eslint errors on your IDE
- [ ] `yarn lint`

### CODE REVIEW

**Comment by PeerRich** [comment]
this is really cool

**Review by PeerRich — APPROVED** [review]

File: packages/eslint-plugin/src/rules/my-first-rule.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import type { Rule } from "eslint";
```
Comment (baileypumfleet) [inline packages/eslint-plugin/src/rules/my-first-rule.ts:1]: Do we still need this my first rule?

**Review by pumfleet — DISMISSED** [review]
Will block this until we get confirmation on if the example rule should still be included in here

File: packages/eslint-plugin/src/rules/my-first-rule.ts (hunk 1 above)
Comment (zomars) [reply packages/eslint-plugin/src/rules/my-first-rule.ts:1]: I've kept it for reference/template but we can remove it. It's not enabled anyways.

**Pull request merged**